homeassistant/components/diagnostics/util.py

```diff
@@ -10,19 +11,23 @@
 
 
 @callback
-def async_redact_data(data: Mapping, to_redact: Iterable[Any]) -> dict[str, Any]:
+def async_redact_data(data: Mapping, to_redact: Iterable[Any]) -> Mapping[Any, Any]:
```

Comment:
This is not true, you can now also return a list. And I guess it already was wrong because it would `return data`. Return value should be set with a generic to same output as data.

```python
T = TypeVar("T")

def async_redact_data(data: T, to_redact: Iterable[Any]) -> T:
```